certora/specs/HashConsensus.spec

```diff
@@ -0,0 +1,742 @@
+/**************************************************
+ *               Methods Declaration              *
+ **************************************************/
+methods {
+    submitConsensusReport(bytes32 report, uint256 refSlot, uint256 deadline) => CONSTANT
+    getLastProcessingRefSlot() returns (uint256) => CONSTANT
+    getConsensusVersion() returns (uint256) => CONSTANT
+
+    reportReceivedCounter(uint256 refSlot, address member, bytes32 report) => getSuccessfulReportSubmitEventInfo(refSlot, member, report)
+}
+
+/**************************************************
+ *                GHOSTS AND HOOKS                *
+ **************************************************/
+// the three ghost variables below store info about a successful report submit
+ghost uint256 ghostReportRefSlot {
+    init_state axiom ghostReportRefSlot == 0;
+}
+ghost address ghostReportMember {
+    init_state axiom ghostReportMember == 0;
+}
+ghost bytes32 ghostReportHash {
+    init_state axiom ghostReportHash == 0;
+}
+
+
+/**************************************************
+ *                CVL FUNCS & DEFS                *
+ **************************************************/
+function saneTimeConfig() returns uint256 {         // returns the timestamp when the valid setting are set
+    env e0; calldataarg args0;
+
+    uint256 correctInitialEpoch;
+    updateInitialEpoch(e0, correctInitialEpoch);    // must be called after constructor
+
+    uint256 slotsPerEpoch; uint256 secondsPerSlot; uint256 genesisTime;
+    slotsPerEpoch, secondsPerSlot, genesisTime = getChainConfig(e0);
+    require slotsPerEpoch == 32;                    // simplification, must be required at constructor
+    require secondsPerSlot == 12;                   // simplification, must be required at constructor
+    require genesisTime < e0.block.timestamp;       // safe assumption, must be required at constructor
+
+    uint256 initialEpoch; uint256 epochsPerFrame; uint256 fastLaneLengthSlots;
+    initialEpoch, epochsPerFrame, fastLaneLengthSlots = getFrameConfig(e0);
+    // uint256 farFutureEpoch = computeEpochAtTimestamp(e0, 0xFFFFFFFFFFFFFFFF); // type(uint64).max
+    // require initialEpoch == farFutureEpoch;         // as in constructor (not good! after constructor updateInitialEpoch() should be called)
+    require epochsPerFrame > 0;                     // constructor already ensures this
+    require epochsPerFrame < 31536000;              // assuming less than 1 year per frame
+    require epochsPerFrame == 86400;                // simplification: Frame = 1 day = 24 * 60 * 60 seconds
+
+    require initialEpoch < ((e0.block.timestamp - genesisTime) / (secondsPerSlot * slotsPerEpoch)); // sane configuration of initialEpoch
+
+    return e0.block.timestamp;
+}
+
+// A helper function to get event info of successful report submit
+function getSuccessfulReportSubmitEventInfo(uint256 refSlot, address member, bytes32 report) returns uint256 {
```

Comment:
This is the trigger summarization of the emit event you've created.
What happens if this trigger is called twice with different arguments?
Your require statements for the ghosts will enforce the arguments to be same (or vacuity).

Reply:
In general, only if a report was submitted successfully two times in a row and I need to check this case, will it matter. But I don't have such a rule or check so I believe the current use is OK.